### ISSUE TITLE
Support more formats

### ISSUE THREAD

**Issue by mifi** [mifi/lossless-cut#68]
Since ffmpeg supports a lot of formats, and Chomium video player does not support as many, I think one possible solution would be to automatically losslessly reformat opened files to a supported format into a temp directory and play back that file instead. Should be relatively easy to implement, but it may be very slow and require a lot of space for big files.

Another possibility would be to serve a custom HTTP endpoint for the video player which automatically re-formats videos on demand, supporting HTTP seeking too, using ffmpeg. This is a bit more complicated unless someone knows an existing implementation of this.

**Comment by wyoung** [comment]
The professional video broadcast/IPTV world still uses MPEG-TS far more than ISO Media (a.k.a. "MP4") and its predecessors. Despite MP4 appearing to take over the world, TS files are not going away any time soon.

Packaging H.264 and either AAC or AC3 in MPEG-TS is increasingly common in the broadcast world. Satellite TV commonly uses this, for example, as does Blu-ray. In this case, you can avoid doing a background transcode by doing a background remux instead, which is much quicker.

As a bonus, this may speed up operation on files accessed over a LAN link: remuxing over the LAN, operating on a local proxy, then generating the output from the local remuxed copy might be faster than operating on a remote MP4 file.

**Comment by mifi** [comment]
> In this case, you can avoid doing a background transcode by doing a background remux instead, which is much quicker.

Yes that's what I was thinking. But we will still have the problem that there will be a potentially huge tmp file created before playback can start.

**Comment by paravorheim** [comment]
While waiting for potential enhancement using ffmpeg and temp files, the PlayMemories application can trim XAVC video losslessly (as far as I can tell). Only problem I have found with this alternative is that the trimming is not as granular as LosslessCut.

**Comment by mifi** [comment]
I have now implemented pre-encoding/pre-formatting of opened videos as of ef54e1b29fabbaf6a628dc3e140a03abc001f64a. Will be included in next release.
